### PR TITLE
Order for resources

### PR DESCRIPTION
**Pull request by cedricmessiant**
Rewrite Resource center admin with react-final-form, user can now change order of the resources

### CODE REVIEW

File: assembl/static2/js/app/components/administration/resourcesCenter/index.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  //   this.props.router.setRouteLeaveHook(this.props.route, null);
+  // }
+
+  // routerWillLeave = () => {
+  //   if (this.dataHaveChanged() && !this.state.refetching) {
+  //     return I18n.t('administration.confirmUnsavedChanges');
+  //   }
+
+  //   return null;
+  // };
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/administration/resourcesCenter/index.jsx]: Remove this commented code once you redo it for react-final-form.

File: assembl/static2/js/app/components/administration/survey/validate.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { I18n } from 'react-redux-i18n';
 
-function langstringIsEmpty(ls) {
+export function langstringIsEmpty(ls) {
   return !ls || Object.values(ls).every(s => s.length === 0);
 }
```
Comment (vincentfretin) [inline assembl/static2/js/app/components/administration/survey/validate.js]: You should put this function in form/utils.js or something.

File: assembl/static2/js/app/graphql/fragments/Resource.graphql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  image {
+    externalUrl
+    mimeType
+    title
+  }
+  text(lang: $lang)
+  title(lang: $lang)
+  titleEntries {
+    localeCode
+    value
```
Comment (vincentfretin) [inline assembl/static2/js/app/graphql/fragments/Resource.graphql]: You can use the langStringEntry fragment here.

File: assembl/static2/js/app/graphql/mutations/updateResource.graphql [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
   $id: ID!
   $image: String
+  $lang: String
   $titleEntries: [LangStringEntryInput]!
   $textEntries: [LangStringEntryInput]
   $embedCode: String
+  $order: Float
 ) {
   updateResource(
     doc: $doc
```
Comment (vincentfretin) [inline assembl/static2/js/app/graphql/mutations/updateResource.graphql]: Don't you need to add lang variable here?

File: assembl/static2/tests/unit/components/form/__snapshots__/multilingualTextFieldAdapter.spec.jsx.snap [hunk 5]
```diff
@@ -32,14 +43,10 @@ exports[`MultilingualTextFieldAdapter component should render a text field touch
     onChange={[Function]}
     onFocus={[Function]}
     placeholder="Title"
+    required={false}
```
Comment (vincentfretin) [inline assembl/static2/tests/unit/components/form/__snapshots__/multilingualTextFieldAdapter.spec.jsx.snap]: Is this normal that this test uses now a not required?

File: assembl/static2/js/app/components/administration/resourcesCenter/index.jsx (hunk 1 above)
Comment (cedricmessiant) [reply assembl/static2/js/app/components/administration/resourcesCenter/index.jsx]: Thanks for the reminder ;-)

File: assembl/static2/js/app/components/administration/survey/validate.js (hunk 2 above)
Comment (cedricmessiant) [reply assembl/static2/js/app/components/administration/survey/validate.js]: done

File: assembl/static2/js/app/graphql/mutations/updateResource.graphql (hunk 4 above)
Comment (cedricmessiant) [reply assembl/static2/js/app/graphql/mutations/updateResource.graphql]: no, it is used only for mutation result (`Resource` fragment)

File: assembl/static2/js/app/graphql/fragments/Resource.graphql (hunk 3 above)
Comment (cedricmessiant) [reply assembl/static2/js/app/graphql/fragments/Resource.graphql]: thanks

File: assembl/static2/js/app/graphql/mutations/updateResource.graphql (hunk 4 above)
Comment (vincentfretin) [reply assembl/static2/js/app/graphql/mutations/updateResource.graphql]: So is this really required in the createResource mutation?

File: assembl/static2/js/app/components/administration/resourcesCenter/index.jsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+                      label={I18n.t('administration.resourcesCenter.imageLabel')}
+                    />
+                    <Field
+                      name={`${name}.doc`}
+                      component={FileUploaderFieldAdapter}
+                      label={I18n.t('administration.resourcesCenter.documentLabel')}
+                    />
+                  </React.Fragment>
+                )}
+                titleMsgId="administration.resourcesCenter.editResourceFormTitle"
```
Comment (MilanBarande) [inline assembl/static2/js/app/components/administration/resourcesCenter/index.jsx]: I think the jsx is easier to read if you put the renderFields prop as the last prop of the component

Comment (cedricmessiant) [reply assembl/static2/js/app/components/administration/resourcesCenter/index.jsx]: you're right, thanks

File: assembl/static2/js/app/components/administration/resourcesCenter/validate.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+// @flow
+import { I18n } from 'react-redux-i18n';
+
+import { i18nValueIsEmpty } from '../../form/utils';
+import type { ResourcesValues } from './types.flow';
+
+function validateResource(resource) {
+  const errors = {};
+  if (i18nValueIsEmpty(resource.title)) {
```
[the remark this answers is not in the document]
Comment (cedricmessiant) [reply assembl/static2/js/app/components/administration/resourcesCenter/validate.js]: I think so. Do you think about something else?

**Pull request merged**